scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -2455,7 +2461,13 @@ void BPF_STRUCT_OPS(lavd_runnable, struct task_struct *p, u64 enq_flags)
 	if (!is_wakeup_ef(enq_flags))
 		return;
 
+	/*
+	 * Filter out unrelated tasks.
+	 */
 	waker = bpf_get_current_task_btf();
+	if (!are_related_tasks(p, waker))
```

Comment:
You can reliably detect whether `@p` is being waken up by the `current` task by testing `SCX_ENQ_WAKEUP` in `@enq_flags`.

Reply:
Hmm... it is strange. I already check SCX_ENQ_WAKEUP in is_wakeup_ef(). However, I still see irrelevant pairs of {`@p`, `current`} are with the SCX_ENQ_WAKEUP flag. Is there any possibility that the kernel config might affect? I am using the CachyOS kernel.

Reply:
AFAICS, `SCX_ENQ_WAKEUP` can only be set when `ops.enqueue()` is being called from `try_to_wakeup()`. Maybe there is a bug in the direct dispatch code and we're leaking flags across enqueues but I don't immediately see how that could happen.

Reply:
Hmm... I will merge this PR to the main for now. Then will further investigate why we are getting `SCX_ENQ_WAKEUP` unexpectedly.